### PR TITLE
Adding cloudstackEntry to enable marshalling cloudstack yamls

### PR DESCRIPTION
**Pull request by maxdrib**
*Issue #, if available:*

*Description of changes:*
This PR was implemented at the suggestion of @jiayiwang7 to enable marshalling of cloudstack yamls, and it uncovered a broken unit test. The implementation is heavily based on VSphere's

*Testing (if applicable):*
Unit tests pass

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: pkg/cluster/config.go [hunk 1]
```diff
@@ -5,22 +5,28 @@ import (
...
-	DockerDatacenter      *anywherev1.DockerDatacenterConfig
-	SnowDatacenter        *anywherev1.SnowDatacenterConfig
-	VSphereMachineConfigs map[string]*anywherev1.VSphereMachineConfig
-	SnowMachineConfigs    map[string]*anywherev1.SnowMachineConfig
-	OIDCConfigs           map[string]*anywherev1.OIDCConfig
-	AWSIAMConfigs         map[string]*anywherev1.AWSIamConfig
-	GitOpsConfig          *anywherev1.GitOpsConfig
-	FluxConfig            *anywherev1.FluxConfig
+	Cluster                  *anywherev1.Cluster
+	CloudStackDatacenter     *anywherev1.CloudStackDatacenterConfig
```
Comment (jiayiwang7) [inline pkg/cluster/config.go:9]: can we add unit test in `parse_test.go`?

Comment (maxdrib) [reply pkg/cluster/config.go:9]: done

**Review by jiayiwang7 — APPROVED** [review]

**Pull request merged**